pkg/setting/setting_plugins.go

```diff
@@ -34,13 +36,59 @@ func extractPluginSettings(sections []*ini.Section) PluginSettings {
 var (
 	defaultPreinstallPlugins = map[string]InstallPlugin{
 		// Default preinstalled plugins
-		"grafana-lokiexplore-app":      {"grafana-lokiexplore-app", "", ""},
-		"grafana-pyroscope-app":        {"grafana-pyroscope-app", "", ""},
-		"grafana-exploretraces-app":    {"grafana-exploretraces-app", "", ""},
-		"grafana-metricsdrilldown-app": {"grafana-metricsdrilldown-app", "", ""},
+		"grafana-lokiexplore-app":      {ID: "grafana-lokiexplore-app"},
+		"grafana-pyroscope-app":        {ID: "grafana-pyroscope-app"},
+		"grafana-exploretraces-app":    {ID: "grafana-exploretraces-app"},
+		"grafana-metricsdrilldown-app": {ID: "grafana-metricsdrilldown-app"},
 	}
 )
 
+func (cfg *Cfg) migrateInstallPluginsToPreinstallPluginsSync(rawInstallPlugins, installPluginsForce string, preinstallPluginsSync map[string]InstallPlugin) {
+	if strings.ToLower(installPluginsForce) == "true" || rawInstallPlugins == "" {
+		return
+	}
+	installPluginsEntries := strings.Split(rawInstallPlugins, ",")
+
+	// Format 1: ID only (e.g., "grafana-clock-panel")
+	// Format 2: ID with version (e.g., "grafana-clock-panel 1.0.1")
+	// Format 3: URL with folder (e.g., "https://grafana.com/api/plugins/grafana-clock-panel/versions/latest/download;grafana-clock-panel")
+	pluginRegex := regexp.MustCompile(`(?:([^;]+);)?([^;\s]+)(?:\s+(.+))?`)
+	for _, entry := range installPluginsEntries {
+		trimmedEntry := strings.TrimSpace(entry)
+		if trimmedEntry == "" {
+			continue
+		}
+
+		matches := pluginRegex.FindStringSubmatch(trimmedEntry)
+
+		if matches == nil {
+			// No match found at all, skip this entry
```

Comment:
I'd log something here, at least in `debug` level, to help debug issues with the regex